__tests__/app.test.js

```diff
@@ -105,61 +105,131 @@ describe("/api/articles", () => {
           expect(comment.comment_id).toBeNumber();
           expect(comment.votes).toBeNumber();
-          expect(comment.created_at).toBeString()
+          expect(comment.created_at).toBeString();
           expect(comment.author).toBeString();
           expect(comment.body).toBeString();
           expect(comment.article_id).toBeNumber();
-        })
-    })
-  })
+        });
+      });
+  });
   it("GET 400: response when given invalid article id", () => {
     return request(app)
       .get("/api/articles/not-an-id/comments")
       .expect(400)
       .then(({ body }) => {
-        const { message } = body
-        expect(message).toBe("bad request")
-    })
-  })
-  it("GET 404: when given valid id but no comments to be returned.", () => {
+        const { message } = body;
+        expect(message).toBe("bad request");
+      });
+  });
+  it("GET 200: responds with empty array when given valid id but no comments to be returned.", () => {
     return request(app)
       .get("/api/articles/2/comments")
-      .expect(404)
+      .expect(200)
+      .then(({ body:{comments} }) => {
+        expect(comments).toEqual([]);
+      });
+  });
+  it("POST 201: when provided with a comment.", () => {
+    const comment = {
+      username: "butter_bridge",
+      body: "Did not look like a pug!",
+    };
+    return request(app)
+      .post("/api/articles/3/comments")
+      .send(comment)
+      .expect(201)
+      .then(({ body: { comment } }) => {
+        expect(comment).toMatchObject({
+          comment_id: 19,
+          votes: 0,
+          body: "Did not look like a pug!",
+          created_at: expect.any(String),
+          author: "butter_bridge",
+          article_id: 3,
+        });
+      });
+  });
+  it("POST 400: when given comment with values in the incorrect data type", () => {
+    const comment = {
+      username: 123,
+      body: "Did not look like a pug!",
+    };
+    return request(app)
+      .post("/api/articles/3/comments")
+      .send(comment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("bad request");
+      });
+  });
+  it("POST 400: when given comment with missing properties", () => {
+    const comment = {
+      body: "Did not look like a pug!",
+    };
+    return request(app)
+      .post("/api/articles/3/comments")
+      .send(comment)
+      .expect(400)
       .then(({ body }) => {
-        expect(body.message).toBe("not found")
-      })
-    
+        expect(body.message).toBe("bad request");
+      });
+  });
+  it("POST 400: when given comment with incorrect properties", () => {
+    const comment = {
+      body: "Did not look like a pug!",
+      votes:10
+    };
+    return request(app)
+      .post("/api/articles/3/comments")
+      .send(comment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toBe("bad request");
+      });
+  });
+  it("POST 400: when given username which doesn't exist", () => {
```

Comment:
If a username doesn't exist but _could_ one day exist then this would be a 404 :) 

A 400 is for entries which are invalid and could never exist without changing the database architecture 👍